gadgets/trace_tcpdrop/program.bpf.c

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// Based on tcpdrop(8) from BCC
+//
+// Copyright 2018 Netflix, Inc.
+// 30-May-2018    Brendan Gregg   Created this.
+// 15-Jun-2022    Rong Tao        Add tracepoint:skb:kfree_skb
+// Copyright 2023 Microsoft Corporation
+
+#include <vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#include <gadget/buffer.h>
+#include <gadget/types.h>
+#include <gadget/macros.h>
+
+#define GADGET_TYPE_TRACING
+#include <gadget/sockets-map.h>
+
+#define TASK_COMM_LEN 16
+
+struct event {
+	struct gadget_l4endpoint_t src;
+	struct gadget_l4endpoint_t dst;
+
+	gadget_timestamp timestamp;
+	__u8 state;
```

Comment:
We should use an enum so the gadget outputs the strings such as "established". The kernel defines the enum but unfortunately without giving a name:
```
$ sudo bpftool btf dump id 1 format c
enum {
        TCP_ESTABLISHED = 1,
        TCP_SYN_SENT = 2,
        TCP_SYN_RECV = 3,
        TCP_FIN_WAIT1 = 4,
        TCP_FIN_WAIT2 = 5,
        TCP_TIME_WAIT = 6,
        TCP_CLOSE = 7,
        TCP_CLOSE_WAIT = 8,
        TCP_LAST_ACK = 9,
        TCP_LISTEN = 10,
        TCP_CLOSING = 11,
        TCP_NEW_SYN_RECV = 12,
        TCP_MAX_STATES = 13,
};
```
See also https://github.com/torvalds/linux/blob/v6.2/include/net/tcp_states.h#L12-L27

So I suggest to redefine the same enum in this gadget but with a name so you can declare the `state` field  with the enum type here.

Reply:
Thanks for mentioning this! Done ✅